src/main/java/Parser.java

```diff
@@ -0,0 +1,103 @@
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Scanner;
+
+/**
+ * Represents the class used to read data from the file and process it.
+ */
+public class Parser {
+
+    public Parser() {
+
+    }
+
+    /**
+     * Contains the methods to read the input and process it into the form required for the algorithm.
+     *
+     * @param s The information to be read and processed
+     * @param t The list where the information is stored after being processed
+     * @throws ParseException If date for event is not in the specified format i.e. MM/dd/yyyy HH:mm
+     */
+    public void readTask(String s, TaskList t) throws ParseException {
+        DateFormat df = new SimpleDateFormat("E, MMM dd yyyy HH:mm");
```

Comment:
It may be better to abstract the code to repopulate your `TaskList` from the file into a method in your `Storage` class. That way if the form of storage changes, it will be relatively easier to just change code in the `Storage` class.